core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -242,10 +242,21 @@ class TransactionCoordinator(brokerId: Int,
         case Ongoing =>
           // indicate to abort the current ongoing txn first. Note that this epoch is never returned to the
           // user. We will abort the ongoing transaction and return CONCURRENT_TRANSACTIONS to the client.
-          // This forces the client to retry, which will ensure that the epoch is bumped a second time. In
-          // particular, if fencing the current producer exhausts the available epochs for the current producerId,
-          // then when the client retries, we will generate a new producerId.
-          Right(coordinatorEpoch, txnMetadata.prepareFenceProducerEpoch())
+          // This forces the client to retry, which will ensure that the epoch is bumped a second time. If the
```

Comment:
I want to be clear about the symptom which is we don't handle epoch exhaustion during the fencing right? If yes, we should mention it more specifically as frequent retries just expose the problem, but the fix itself suggests to an error handling gap.

Reply:
Tried to clarify the reasoning a bit better in the comment. I also moved part of it to a separate method where I extracted the fencing logic. Let me know if it's clearer.